Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftWindowsEventFirewall.yaml

```diff
@@ -218,40 +250,46 @@ ParserQuery: |
             DstHostname = iff(isOutBound, "", DvcHostname),
             SrcHostname = iff(isOutBound, DvcHostname, "")
             // *************** Postfilterring *****************************************************************
-          | where  (isnull(dstportnumber) or DestPort == dstportnumber ) 
-              and  (array_length(srcipaddr_has_any_prefix)==0 
-                    or has_any_ipv4_prefix(SrcIpAddr ,srcipaddr_has_any_prefix)
-                  ) 
-              and (array_length(dstipaddr_has_any_prefix)==0 
-                    or has_any_ipv4_prefix(DstIpAddr,dstipaddr_has_any_prefix)
-                  ) 
+          | where  (isnull(dstportnumber) or DestPort == dstportnumber )
           // *************** / Postfilterring *****************************************************************
       };
   //////////////////////////////////////////////////////
   // this query extract the data from WindowsEvent table
   //////////////////////////////////////////////////////
   let WindowsFirewall_WindowsEvent=(starttime:datetime=datetime(null), endtime:datetime=datetime(null)
-  , srcipaddr_has_any_prefix:dynamic=dynamic([]), dstipaddr_has_any_prefix:dynamic=dynamic([]), dstportnumber:int=int(null)
+  , srcipaddr_has_any_prefix:dynamic=dynamic([]), dstipaddr_has_any_prefix:dynamic=dynamic([]), ipaddr_has_any_prefix:dynamic=dynamic([]),dstportnumber:int=int(null)
   , hostname_has_any:dynamic=dynamic([]), dvcaction:dynamic=dynamic([]),eventresult:string='*', disabled:bool=false
-   ){   
+   ){ 
+      let src_or_any=set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let dst_or_any=set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let ip_any =set_union(srcipaddr_has_any_prefix, dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
       WindowsEvent | where not(disabled)
       | where EventID between (5150 .. 5159)
```

Comment:
Unrelated to this change, but can you check why the SecurityEvent version has so many variants and WindowsEvent is so much simpler?

Reply:
We can look into it, but from a glance: windows event has the dynamic EventData in already parsed format